src/pages/Admin/Charity/Permissions/MobileTable.tsx

```diff
@@ -0,0 +1,127 @@
+import { Disclosure } from "@headlessui/react";
+import { useFormContext } from "react-hook-form";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useGetWallet } from "contexts/WalletContext";
+import { DrawerIcon } from "components/Icon";
+import { CheckField, Field } from "components/form";
+import { FormValues } from "./schema";
+
+export default function MobileTable({ className = "", disabled = false }) {
+  const { endow_type } = useAdminResources<"charity">();
+  const { wallet } = useGetWallet();
+  const { getValues, register, watch } = useFormContext<FormValues>();
+
+  const { initialValues, ...formValues } = getValues();
```

Comment:
nit:  though `getValues` doesn't trigger rerender, it's quite uintuitive that `formFields` are rendered by `formValues` which it also updates
`formValues` -> `formFields` -> `formValues`

might be better to just create static `formFields` directly from FormValues
```ts
const formFields:{K in keyof FormValues: K } = {/** ... */}
const fieldNames = Object.keys(formFields)
```

Reply:
Sure.